### DISCUSSION TITLE
Re-creating the instance

### DISCUSSION THREAD

**Discussion by osrl** [InsertKoinIO/koin#1574] [Q&A] [answered]
I have a bean definition defined like this:
`single { MyClass(SOME_CLIENT_ID) }`

And some other single beans also depends on it.

I want to re-create all instances depending on `MyClass` when I change SOME_CLIENT_ID.

Can anyone help me with this?

**Comment by osrl** [comment, marked as answer]
I used `unloadKoinModules` and `loadKoinModules` for this.

**Reply by arnaudgiuliani** [reply]
yep, good choice 👍
